### PR TITLE
Named note

### PR DESCRIPTION
**Pull request by NeroBlackstone**
Related [issue](https://github.com/JuliaMusic/MIDI.jl/issues/165)

### CODE REVIEW

File: src/scale.jl [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using MIDI
 using StatsBase
+using DataStructures
```
Comment (Datseris) [inline src/scale.jl:3]: You need to add DataStructures to Project.toml. Also need to add a compat version for it.

**Review by Datseris — COMMENTED** [review]
Thanks for taking care of also updating the code base. I've left some comments to improve NamedNote. Note that tests fail. I suspect its because you use DataStructures but haven't added it to Project.toml.

File: src/namednote.jl [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    chpr = note.channel == 0 ? "" : " | channel $(note.channel)"
+    velprint = rpad("vel = $(Int(note.velocity))", 9)
+    print(io, "NamedNote $nn | $velprint | "*
+    "pos = $(Int(note.position)), "*
+    "dur = $(Int(note.duration))"*chpr)
+end
+
+import Base.==
+==(n1::NamedNote, n2::NamedNote) =
+    name_to_pitch(n1.name) == name_to_pitch(n2.name) &&
```
Comment (NeroBlackstone) [inline src/namednote.jl:86]: Do you think this is the right way to compare two NamedNote?

Comment (Datseris) [reply src/namednote.jl:86]: Yes.

File: Project.toml [hunk 3]
```diff
@@ -19,6 +19,7 @@ MIDI = "1.4.1, 2.0"
 MotifSequenceGenerator = "1.0"
 Reexport = "0.2, 0.3, 1"
 StatsBase = "0.32, 0.33"
+DataStructures = "0.18"
```
[the remark this answers is not in the document]
Comment (NeroBlackstone) [reply Project.toml:22]: I have added it, seems a test failed only on macos.
Maybe caused by my old code base update.

File: src/namednote.jl [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        end
+end
+
+NamedNote(pitch_name::String; position = 0, velocity = 100, duration = 960, channel = 0) = 
+    NamedNote(pitch_name, velocity, position, duration, channel)
+
+NamedNote(n::Note; pitch_name::String = "") = 
+    length(pitch_name) == 0 ? NamedNote(pitch_to_name(n.pitch), n.position, n.velocity, n.duration, n.channel) : NamedNote(name, n.position, n.velocity, n.duration, n.channel) 
+
+NamedNotes(notes_string::String; tpq::Int = 960) = Notes([NamedNote(String(s)) for s in split(notes_string," ")], tpq)
```
Comment (Datseris) [inline src/namednote.jl:71]: This function does not have a documentation string!

File: test/namednote_test.jl [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    @test NamedNote("Db5") == NamedNote("C#5")
+
+    nns = NamedNotes("C#6 Db5 E4")
+    @test nns[1].name == "C♯6"
+    @test nns[2].name == "D♭5"
+    @test nns[3].name == "E4"
+    
+    cn = Note("C#4")
+    nn3 = NamedNote(cn)
+    @test nn3.name == "C♯4"
```
Comment (Datseris) [inline test/namednote_test.jl:19]: This test is unecessary, you already do the same test in the start of the test suite. Instead, write a test that compares a `NamedNote` with a `Note` generated from it and vice versa.

**Comment by Datseris** [comment]
This test failed on windows now: https://github.com/JuliaMusic/MusicManipulations.jl/actions/runs/5146154049/jobs/9264791892?pr=65#step:7:172 and before that it failed on MacOS. Something weird is going on here...

**Comment by NeroBlackstone** [comment]
Commit nothing try to re-run ci test, and all passed.
I think some bugs in data_extraction test, but I'm not sure it is caused by my code change.

**Comment by NeroBlackstone** [comment]
> Great, seems good. Can you please also add an entry in the CHANGELOG.md file?

Done! 

If you have more time please help me review this [issue](https://github.com/JuliaMusic/MIDI.jl/issues/164), Thank you very much.

**Pull request merged**